doc/doxygen-root/doxygen-markdown/markdown.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -euo pipefail
+
+BINDIR=$(dirname $(realpath ${BASH_SOURCE[0]}))
+FILES=$(find . -name '*.md')
+
+# Append the last modified date to the end of every markdown file
+
+echo
+echo "Appending last modified dates to markdown files"
+$BINDIR/append-last-modified-dates.py $FILES
+
+# Doxygen parses incorrectly a link [what a link](what-a-link.md) that
+# is broken over two lines.
+# Doxygen requires that headings '# heading' have labels '{# heading}'
+# for section linking to work.  The markdown extension "php Markdown
+# Extra" supports section labels.
+# Use pandoc to remove line breaks from paragraphs and to output a
+# markdown extension with section labels.
+# Note: Need to read markdown as markdown_phpextra and not default
+# markdown to preserve doxygen pragmas like \ingroup.
+
+# Bug: This is currently interacting badly with \dot in cprover markdown
```

Comment:
To what extent should we be concerned about this? Is there any plan to do something about this?

Reply:
This preprocessing is required on the cprover-manual pages as long as they are served up by javascript on prover.org.  I did a quick scan of the architectural pages, and I saw no obvious problems caused by omitting this preprocessing on the current set of pages.  But I think doxygen's brittle treatment of markdown files is a potential issue, and I think we would be better off just deciding to write doxygen-friendly markdown and just living with subpar rendering by other tools (like on github).